src/services/file.service.ts

```diff
@@ -2,240 +2,53 @@
 import { type Pain001 } from '../classes/pain.001.001.11';
 import * as fs from 'fs';
 import * as readline from 'readline';
-import { v4 as uuidv4 } from 'uuid';
-import { dbService } from '..';
+import { databaseClient, dbService } from '..';
 import { configuration } from '../config';
 import { LoggerService } from '../logger.service';
 import {
   GetPacs002,
   GetPacs008,
   GetPain013,
+  GetPain001FromLine,
 } from './message.generation.service';
 import { executePost } from './utilities.service';
 import { handleTransaction } from './save.transactions.service';
 import { type Pain013 } from '../classes/pain.013.001.09';
 import { type Pacs008 } from '../classes/pacs.008.001.10';
 
-export const GetPain001FromLine = (columns: string[]): Pain001 => {
-  const end2endID = columns[2];
-  const testID = uuidv4().replace('-', '');
+const getMissingTransaction = async (
+  batchSourceFileLine: readline.Interface,
+): Promise<string[]> => {
+  let counter = 0;
+  let endToEndIds: string[] = [];
+  for await (const line of batchSourceFileLine) {
+    if (counter === 0) {
+      counter++;
+      continue;
+    }
```

Comment:
Not a fan of this, can we not initialise the counter as 1?

Reply:
Still adding the first transaction from the source file which is headers and not exactly a transaction

Reply:
Then slice it?